File: leet-code/decompressRLElist.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var decompressRLElist = function (nums) {
  let result = []
  for (let i = 0; i < nums.length; i += 2) {
    result = result.concat(Array(nums[i]).fill(nums[i + 1], 0, nums[i]))
  }
  return result
}

console.log(decompressRLElist([1, 2, 3, 4]))

File: leet-code/decompressRLElist.go
package main

func decompressRLElist(nums []int) []int {
	var result []int
	for i := 0; i < len(nums); i += 2 {
		for j := 0; j < nums[i]; j++ {
			result = append(result, nums[i+1])
		}
	}
	return result
}
